hummingbot/strategy/pure_market_making/pure_market_making.pyx

```diff
@@ -737,6 +747,16 @@ cdef class PureMarketMakingStrategy(StrategyBase):
                                           f"making may be dangerous when markets or networks are unstable.")
 
             proposal = None
+            ref_price_available = not self.get_price().is_nan()
+            if not ref_price_available:
+                self.logger().info("Reference price not available. Waiting for a new price to resume strategy...")
+
+                if len(self.active_orders) > 0:
+                    self.logger().info("Cancelling any active orders for safety...")
+                    self.c_cancel_all_active_orders()
+
+                return
```

Comment:
there's this statement at the end of this function:
```python
        finally:
            self._last_timestamp = timestamp
```
That won't get executed if we short-circuit the return here. Is this a problem?

Reply:
good catch.. this seems to be used for checking if the logger should report warnings based on some interval:
https://github.com/mento-protocol/hummingbot/pull/1/files#diff-ff54f4274a270c937744102bfad74438719543f0f63cf3a45622e62877cfbb30L718-L721